webroot/src/pages/LicensingDetail/LicensingDetail.ts

```diff
@@ -19,12 +21,66 @@ export default class LicensingDetail extends Vue {
     //
     // Lifecycle
     //
+    async mounted() {
+        await this.fetchLicenseeData();
+
+        if (!this.licensee) {
+            this.$router.push({ name: '404' });
+        }
+    }
 
     //
     // Computed
     //
+    get licenseStore(): any {
+        return this.$store.state.license;
+    }
+
+    get licenseeId(): string {
+        const routeId: string = this.$route.params.licenseeId as string || '';
+
+        return routeId;
+    }
+
+    get licensee(): Licensee | null {
+        const { licenseeId } = this;
+        let storeRecord: Licensee | null = null;
+
+        if (licenseeId && this.licenseStore.model) {
```

Comment:
Fair question. The main reasoning is because `licenseeId` is reused on the following line. Generally we'll try cache vars that are used more than once, both for readability & also avoid the potential cost of traversing the parent object multiple times.

Reply:
Fair enough. As long as we're consistent, this is good with me!